sql/moz-fx-data-shared-prod/telemetry/active_users/view.sql

```diff
@@ -0,0 +1,193 @@
+WITH fenix_distribution_id AS (
+  SELECT
+    client_info.client_id AS client_id,
+    sample_id AS sample_id,
+    DATE(submission_timestamp) AS submission_date,
+    ARRAY_AGG(
+      metrics.string.metrics_distribution_id IGNORE NULLS
+      ORDER BY
+        submission_timestamp ASC
+    )[SAFE_OFFSET(0)] AS distribution_id,
+  FROM
+    `moz-fx-data-shared-prod.fenix.baseline`
+  GROUP BY
+    client_id,
+    sample_id,
+    submission_date
+)
+-- Firefox Desktop
+SELECT
+  submission_date,
+  client_id,
+  CASE
+    WHEN isp_name = 'BrowserStack'
+      THEN CONCAT(app_name, ' ', isp_name)
+    WHEN distribution_id = 'MozillaOnline'
+      THEN CONCAT(app_name, ' ', distribution_id)
+    ELSE app_name
+  END AS app_name,
+  days_seen_bits,
+  days_active_bits,
+  CAST(mozfun.bits28.days_since_seen(days_active_bits) = 0 AS BOOLEAN) AS is_dau,
+  CAST(mozfun.bits28.days_since_seen(days_active_bits) < 7 AS BOOLEAN) AS is_wau,
+  CAST(mozfun.bits28.days_since_seen(days_active_bits) < 28 AS BOOLEAN) AS is_mau,
+  CAST(mozfun.bits28.days_since_seen(days_seen_bits) = 0 AS BOOLEAN) AS is_daily_user,
+  CAST(mozfun.bits28.days_since_seen(days_active_bits) < 7 AS BOOLEAN) AS is_weekly_user,
+  CAST(mozfun.bits28.days_since_seen(days_seen_bits) < 28 AS BOOLEAN) AS is_monthly_user,
+  IF(
+    LOWER(isp) <> "browserstack"
+    AND LOWER(distribution_id) <> "mozillaonline",
+    TRUE,
+    FALSE
+  ) AS is_desktop,
+  FALSE AS is_mobile
+FROM
+  `moz-fx-data-shared-prod.telemetry_derived.clients_last_seen_v2`
+UNION ALL
+  -- Fenix
+SELECT
+  submission_date,
+  client_id,
+  CASE
+    WHEN isp = 'BrowserStack'
+      THEN CONCAT('Fenix ', isp)
+    WHEN distribution_id = 'MozillaOnline'
+      THEN CONCAT('Fenix', ' ', distribution_id)
+    ELSE 'Fenix'
+  END AS app_name,
+  days_seen_bits,
+  days_active_bits,
+  CAST(mozfun.bits28.days_since_seen(days_active_bits) = 0 AS BOOLEAN) AS is_dau,
+  CAST(mozfun.bits28.days_since_seen(days_active_bits) < 7 AS BOOLEAN) AS is_wau,
+  CAST(mozfun.bits28.days_since_seen(days_active_bits) < 28 AS BOOLEAN) AS is_mau,
+  CAST(mozfun.bits28.days_since_seen(days_seen_bits) = 0 AS BOOLEAN) AS is_daily_user,
+  CAST(mozfun.bits28.days_since_seen(days_seen_bits) < 7 AS BOOLEAN) AS is_weekly_user,
+  CAST(mozfun.bits28.days_since_seen(days_seen_bits) < 28 AS BOOLEAN) AS is_monthly_user,
+  FALSE AS is_desktop,
+  IF(
+    LOWER(isp) <> "browserstack"
+    AND LOWER(distribution_id) <> "mozillaonline",
+    TRUE,
+    FALSE
+  ) AS is_mobile
+FROM
+  `moz-fx-data-shared-prod.fenix.baseline_clients_last_seen`
+LEFT JOIN
+  fenix_distribution_id
+  USING (client_id, sample_id, submission_date)
+UNION ALL
+-- Firefox iOS
+SELECT
+  submission_date,
+  client_id,
+  CASE
+    WHEN isp = 'BrowserStack'
+      THEN CONCAT('Firefox iOS', ' ', isp)
+    ELSE 'Firefox iOS'
+  END AS app_name,
+  days_seen_bits,
+  days_active_bits,
+  CAST(mozfun.bits28.days_since_seen(days_active_bits) = 0 AS BOOLEAN) AS is_dau,
```

Comment:
This yields an unexpected NULL result when `days_active_bits = 0` :

```sql
SELECT CAST(mozfun.bits28.days_since_seen(0) = 0 AS BOOLEAN)  # NULL
```

This can happen when a user sends a ping but is not active. The expected behavior here is to return `false`.

We could add a coalesce:

```sql
CAST(COALESCE(mozfun.bits28.days_since_seen(days_active_bits) = 0, FALSE) AS BOOLEAN) AS is_dau
```

or a less verbose version:

```sql
COALESCE(mozfun.bits28.days_since_seen(days_active_bits) = 0, FALSE) AS is_dau
```

This change needs to be made for `is_wau` and `is_mau` as well.

Reply:
Good catch, yes it returns NULL due to the SAFE.LOG for 0 and negative values. I'll go for IFNULL since it's only checking one column.